### ISSUE TITLE
Can you use puppeteer instead of selenium as web driver to avoid checkpoint?

### ISSUE THREAD

**Issue by VanHop94** [Schmavery/facebook-chat-api#813]
Hi

Thank for you libs, it's awesome for me. But 1 issue, I usually get a checkpoint. I read the document and see that we use selenium to simulate user behavior. But Facebook can easy to check bot if we using selenium,  puppeteer help us resolve it with https://github.com/berstend/puppeteer-extra/tree/master/packages/puppeteer-extra-plugin-stealth

**Comment by lequanglam** [comment]
actually, we don't. we use `request` to request to Facebook.

**Comment by VanHop94** [comment]
I think the use request is the reason the Facebook account usually checkpoint. I try to use the same account, the same behavior (send message to not friend user),  on the browser and your code. When use browser it works well, but via code, it gets checkpoint :(

**Comment by lequanglam** [comment]
try changing user agent.

**Comment by VanHop94** [comment]
yes, I tried to set user agent the same on my laptop, but it doesn't help

**Comment by 7osCraft** [comment]
Try these things:
- Enable 2FA
- Enable autoMarkDelivery and autoMarkRead
- Don't use a server because Facebook can detect the ip

**Comment by VanHop94** [comment]
I've set user Agent, but sometimes I received an email saying that I log in from an unknown device
